src/Symfony/Bundle/FrameworkBundle/Secrets/DotenvVault.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secrets;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class DotenvVault extends SodiumVault
+{
+    private $dotenvFile;
+
+    public function __construct(string $dotenvFile)
+    {
+        $this->dotenvFile = strtr($dotenvFile, '/', \DIRECTORY_SEPARATOR);
+    }
+
+    public function generateKeys(bool $override = false): bool
+    {
+        $this->lastMessage = 'The dotenv vault doesn\'t encrypt secrets thus doesn\'t need keys.';
+
+        return false;
+    }
+
+    public function seal(string $name, string $value): void
+    {
+        $this->lastMessage = null;
+        $this->validateName($name);
+        $k = $name.'_SECRET';
+        $v = rawurlencode($value);
+
+        $content = file_exists($this->dotenvFile) ? file_get_contents($this->dotenvFile) : '';
+        $content = preg_replace("/^$name=.*/m", "$k=$v", $content, -1, $count);
```

Comment:
pattern should be `/^$k=.*/m`.
and probably either `/^$k=.*/im` or converting the name in uppercase to avoid duplicate variables

Reply:
$name updated - not sure about the `i`: env vars are case sensitive, so why?

Reply:
```
bin/console sec:set FOO
bin/console sec:set foo
```
will generate 2 lines in .env.local it could be confusing

Reply:
That's the price of case sensitiveness. I don't think we want case-insensitivity here (I'm pretty sure this would bite us soon if we did).